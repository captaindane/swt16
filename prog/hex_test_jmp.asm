lil    r1, 0xF      ; r1        <- 0xF
addpci r2, 0x20     ; r2        <- 0x20 + 0x02 (pc) = 0x22
li     r3, 0x22     ; r3        <- 0x22
addpc  r4, r3       ; r4        <- pc + 0x22
sho    r1, 0x0(r0)  ; dmem[0x0] <- 0xF
jal    r5, 0x18     ; r5        <- pc+4 = 0x18
;=====================================================
; We jump over the next two lines: dmem[0x0] = 0xF
;=====================================================
sho r0, 0x0(r0)     ; dmem[0x0] <- 0x0 (not executed)
;=====================================================
sho  r1, 0x2(r0)    ; dmem[0x2] <- 0xF
jalr r6, r2         ; r6        <- pc+2 = 0x1E
;=====================================================
; We jump over the next two lines: dmem[0x2] = 0xF
;=====================================================
sho r0, 0x2(r0)     ; dmem[0x2] <- 0x0 (not executed)
;=====================================================
sho  r1, 0x4(r0)    ; dmem[0x4] <- 0xF
jalr r7, r4         ; r7 <- pc+2 = 0x28
;=====================================================
; We jump over the next two lines: dmem[0x4] = 0xF
;=====================================================
sho r0, 0x4(r0)     ; dmem[0x4] <- 0x0 (not executed)
;=====================================================
sho r5, 0x6(r0)     ; dmem[0x6] <- r5 = 0x14
sho r6, 0x8(r0)     ; dmem[0x8] <- r6 = 0x1E
sho r7, 0xA(r0)     ; dmem[0xA] <- r7 = 0x28
