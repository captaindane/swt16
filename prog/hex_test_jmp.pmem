F411  //lil    r1, 0xF
0521  //addpci r2, 0x20
0020
0331  //li     r3, 0x22
0022
4342  //addpc  r4, r3
0153  //sho    r1, 0x0(r0)
0000
0051  //jal    r5, _label
0018
0053  //sho r0, 0x0(r0)
0000
0153  //sho  r1, 0x2(r0)
0002
1262  //jalr r6, r2
0053  //sho r0, 0x2(r0)
0002
0153  //sho  r1, 0x4(r0)
0004
1472  //jalr r7, r4
0053  //sho r0, 0x4(r0)
0004
0553  //sho r5, 0x6(r0)
0006
0653  //sho r6, 0x8(r0)
0008
0753  //sho r7, 0xA(r0)
000A
