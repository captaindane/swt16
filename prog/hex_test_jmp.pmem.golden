F411 // lil    r1, 0xF      : r1      <- 0xF
0521 // addpci r2, 0x20     : r2      <- 0x20 + 0x02 = 0x22
0020 // ...
0331 // li     r3, 0x22     : r3      <- 0x22
0022 // ...
4342 // addpc  r4, r3       : r4      <- pc + 0x22
0153 // sho    r1, [r0] +0  : dmem[0] <- 0xF
0000 // ...
0051 // jal    r5           : r5      <- pc+4 = 0x18
0018 // ...
//=====================================================
// We jump over the next two lines: dmem[ 0] = 0xF
//=====================================================
0053 // sho r0, [r0] +0  : dmem[0] <- 0x0 (not executed)
0000 // ...                               (not executed)
//=====================================================
0153 // sho r1, [r0] +2  : dmem[2] <- 0xF
0002 // ...
1262 // jalr r6, r2      : r6      <- pc+2 = 0x1E
//=====================================================
// We jump over the next two lines: dmem[ 2] = 0xF
//=====================================================
0053 // sho r0, [r0] +2  : dmem[2] <- 0x0 (not executed)
0002 // ...                               (not executed)
//=====================================================
0153 // sho    r1, [r0] +4  : dmem[4] <- 0xF
0004 // ...
1472 // jalr r7, r4         : r7 <- pc+2 = 0x28
//=====================================================
// We jump over the next two lines: dmem[ 4] = 0xF
//=====================================================
0053 // sho r0, [r0] +4  : dmem[4] <- 0x0 (not executed)
0004 // ...                               (not executed)
//=====================================================
0553 // sho r5, [r0] +6  : dmem[6] <- r5 = 0x14
0006 // ...
0653 // sho r6, [r0] +8  : dmem[8] <- r6 = 0x1E
0008 // ...
0753 // sho r7, [r0] +A  : dmem[A] <- r7 = 0x28
000A // ...
